bigc/api_client.py

```diff
@@ -35,25 +41,39 @@ def request(
 
         self._validate_path(path)
 
-        try:
-            response = requests.request(
-                method,
-                self._prepare_url(path),
-                json=data,
-                params=self._process_params(params),
-                headers=self._get_standard_request_headers() | headers,
-                timeout=timeout,
-            )
-        except requests.Timeout as exc:
-            raise GatewayTimeoutError() from exc
-        except requests.RequestException as exc:
-            raise BigCommerceNetworkError() from exc
-
-        if response.ok:
-            # Return None for empty responses instead of raising
-            return response.json() if response.text else None
-        else:
-            self._handle_error_response(response)
+        url = self._prepare_url(path)
+        params = self._process_params(params)
+        headers = self._get_standard_request_headers() | headers
+
+        def perform_request() -> Any:
+            try:
+                response = requests.request(
+                    method,
+                    url,
+                    json=data,
+                    params=params,
+                    headers=headers,
+                    timeout=timeout,
+                )
+            except requests.Timeout as exc:
+                raise GatewayTimeoutError() from exc
+            except requests.RequestException as exc:
+                raise BigCommerceNetworkError() from exc
+
+            if response.ok:
+                # Return None for empty responses instead of raising
+                return response.json() if response.text else None
+            else:
+                self._handle_error_response(response)
+
+        last_exc: Exception | None = None
+        for try_idx in range(retry_attempts + 1):
+            try:
+                return perform_request()
+            except BigCommerceServerError as exc:
+                last_exc = exc
```

Comment:
We aren't trying to retry for _any_ error, there are specific types of errors we are trying to catch (timeouts, connection problems).

Reply:
This isn't any error, this is any 5xx error

Reply:
https://github.com/MedShift/bigc/pull/55/commits/d687c55695b03eed06a2af212879b573c03ed161